### ISSUE TITLE
Freeze Encoder skipping an iteration on GPU

### ISSUE THREAD

**Issue by svabhishek29** [qubvel/segmentation_models#33]
When i use Unet with resnet 152 with image net weights and encoder freezed, the training on cpu works as intened, but on the gpu the training runs every other itreation, as in 1, 3, 5, 7...it is skipping even iterations. 
Even though the training looks to be going well giving close to 98% acc on binary cross entropy, the predictions are all in 3X10-8 range, basically the predicted images are all 0.
This issue dosen't happen if freeze encoder is not used

**Comment by qubvel** [comment]
Hi @svabhishek29 
Keras has strange Batch Normalization layer, always train whole model to update its weights. Encoder freezing useful for training decoder a few epochs on start.
